### PR TITLE
Address windows build issues

### PR DESCRIPTION
**Pull request by yuslepukhin**
Intro SubCompactionState move functionality
  =delete copy functionality
  #ifdef SyncPoint in tests for Windows Release builds

### CODE REVIEW

File: db/compaction_job.cc [hunk 1]
```diff
@@ -113,7 +113,7 @@ struct CompactionJob::SubCompactionState {
   // is in or beyond the last file checked during the previous call
   std::vector<size_t> level_ptrs;
 
-  explicit SubCompactionState(Compaction* c, Slice* _start, Slice* _end,
+  SubCompactionState(Compaction* c, Slice* _start, Slice* _end,
                            SequenceNumber earliest, SequenceNumber visible,
                            SequenceNumber latest)
```
Comment (siying) [inline db/compaction_job.cc]: Can you fix the indenting?

**Comment by siying** [comment]
I fixed the build before releasing 3.13 in a more ugly way than this: https://reviews.facebook.net/D43647 Maybe we should change to this pattern?

**Comment by yuslepukhin** [comment]
@siying That may cost perf. We want move semantics whenever possible.

**Comment by siying** [comment]
@yuslepukhin I mean change the codes I modified to your way:)

**Comment by yuslepukhin** [comment]
@siying We understood each other :)

**Pull request merged**